src/node.ts

```diff
@@ -243,6 +243,39 @@ class Node<T> {
     }
   }
 
+  /*
+   * Look at only the results from this node (or the empty string node this one
+   * contains) for exact matched of the query.
+   *
+   * We use the passed in pqueue which has a limit and unique flag to
+   * configure the search.
+   */
+  getSortedExactResults(query: string, results: Array<T>, opts: SearchOptions) {
+    const seenKeys = new Set<string>();
+
+    if (!this.leaf) {
+      this.children['']?.getSortedExactResults(query, results, opts);
```

Comment:
Doing a bit of recursion here.  I wasn't too concerned about doing so only because this should only happen a single time.

Reply:
Sounds good. Can you expand on the validateInvarients function to check that if the child's key is the empty string that all the children nodes are leaves?